### PR TITLE
LU decomposition for symmetric and Hermitian static matrices

### PR DESCRIPTION
**Pull request by mateuszbaran**
This is an attempt at solving a part of #971 (specifically, example 1).

### CODE REVIEW

File: src/lu.jl [hunk 1]
```diff
@@ -110,6 +112,9 @@ __lu(A::StaticMatrix{M,0,T}, ::Val{Pivot}) where {T,M,Pivot} =
 __lu(A::StaticMatrix{1,1,T}, ::Val{Pivot}) where {T,Pivot} =
     (SMatrix{1,1}(one(T)), A, SVector(1))
 
+__lu(A::LinearAlgebra.HermOrSym{T,<:StaticMatrix{1,1,T}}, ::Val{Pivot}) where {T,Pivot} =
```
Comment (thchr) [inline src/lu.jl:115]: Out of curiosity, what is the purpose of this additional method? Just a performance optimization?

**Review by thchr — APPROVED** [review]
I read through this PR and it seems reasonable and correct to me.

File: src/lu.jl (hunk 1 above)
Comment (mateuszbaran) [reply src/lu.jl:115]: IIRC yes, that's just an optimization similar to `__lu(A::StaticMatrix{1,1,T}, ::Val{Pivot})`.

**Comment by mateuszbaran** [comment]
I keep getting too many seemingly random CI failures here that need to be resolved before it can be merged.

**Pull request merged**